src/main/java/javaslang/collection/List.java

```diff
@@ -278,6 +278,40 @@
         return List.ofAll(Iterator.ofAll(array));
     }
 
+    /**
+     * Returns a List containing {@code n} values of a given Function {@code f}
+     * over a range of integer values from 0 to {@code n - 1}.
+     *
+     * @param <T> Component type of the List
+     * @param n The number of elements in the List
+     * @param f The Function computing element values
+     * @return A List consisting of elements {@code f(0),f(1), ..., f(n - 1)}
+     * @throws NullPointerException if {@code f} is null
+     */
+    static <T> List<T> tabulate(int n, Function<? super Integer, ? extends T> f) {
+        Objects.requireNonNull(f, "f is null");
+        int nOrZero = java.lang.Math.max(n, 0);
+        @SuppressWarnings("unchecked")
+        T[] elements = (T[]) new Object[nOrZero];
+        for (int i = 0; i < n; i++) {
```

Comment:
oh, we use `n` instead of nOrZero. Please double-check all other places.

Ruslan would overwrite the given method arg like this: `n = Math.max(n, 0);` ;-)